lib/fsm/fsm.go

```diff
@@ -329,13 +329,28 @@ You can redeploy upgrade agents on all cluster nodes using "./gravity agent depl
 }
 
 func (f *FSM) rollbackPhaseRemotely(ctx context.Context, p Params, phase storage.OperationPhase, server storage.Server) error {
-	return f.RunCommand(ctx, f.Runner, server, Params{
+	err := f.RunCommand(ctx, f.Runner, server, Params{
 		PhaseID:  p.PhaseID,
 		Force:    p.Force,
 		Resume:   p.Resume,
 		Rollback: true,
 		Progress: p.Progress,
 	})
+	if err == nil {
+		// if the remote phase rollback is successful, we need to mark it in our local database
+		// because etcd might not be available to synchronize the changes back to us
+		err = f.ChangePhaseState(ctx,
```

Comment:
`gravity rollback` would fail in the middle of rolling back a plan whenever a phase was rolled back remotely. 
```
[vagrant@node-1 installer]$ sudo ./gravity rollback
You are about to rollback the following operation:
Type                 ID                                       State                  Created
----                 --                                       -----                  -------
operation_update     c8e539eb-23ef-4e7b-999c-e022ce04963b     update_in_progress     2020-10-07 01:07
Consider checking the operation plan and using --dry-run flag first to see which actions will be performed.
You can suppress this warning in future by providing --confirm flag.
Proceed? (yes/no):
yes
Wed Oct  7 01:14:20 UTC Rolling back "/masters/node-3/untaint" locally
Wed Oct  7 01:14:24 UTC Rolling back "/masters/node-3/endpoints" locally
Wed Oct  7 01:14:31 UTC Rolling back "/masters/node-3/uncordon" locally
Wed Oct  7 01:14:39 UTC Rolling back "/masters/node-3/taint" locally
Wed Oct  7 01:14:45 UTC Rolling back "/masters/node-3/system-upgrade" on node node-3
        Still rolling back "/masters/node-3/system-upgrade" on node node-3 (10 seconds elapsed)
        Still rolling back "/masters/node-3/system-upgrade" on node node-3 (20 seconds elapsed)
        Still rolling back "/masters/node-3/system-upgrade" on node node-3 (30 seconds elapsed)
        Still rolling back "/masters/node-3/system-upgrade" on node node-3 (40 seconds elapsed)
Wed Oct  7 01:15:31 UTC operation(update(c8e539eb-23ef-4e7b-999c-e022ce04963b), cluster=dev.test, created=2020-10-07 01:07) finished in 1 minute
[ERROR]: phase /masters/node-3/drain cannot be rolled back because some phases that depend on it haven't been rolled back yet. Please rollback the following phases first:
        /masters/node-3/system-upgrade
you can pass --force flag to override this check and force phase /masters/node-3/drain rollback
```
This was because `rollbackPhaseRemotely` was not updating the local plan after rolling back a phase.